evm/src/hosts/EvmHost.sol

```diff
@@ -712,7 +721,8 @@ abstract contract EvmHost is IIsmpHost, IHostManager, Context {
 
         require(metadata.sender != address(0), "Unknown request");
         require(metadata.sender != _msgSender(), "User can only fund own requests");
-        require(IERC20(feeToken()).transferFrom(_msgSender(), address(this), amount), "Payer has insufficient funds");
+
+        _hostParams.permit2.transferFrom(_msgSender(), address(this), uint160(amount), feeToken()); // Payer has insufficient funds
```

Comment:
I think for this we should use `permitTransferFrom`, otherwise the user will have to make 2 txs, one to approve and one to fund their request.

https://docs.uniswap.org/contracts/permit2/reference/signature-transfer#single-permittransferfrom